src/vulkan-renderer/wrapper/shader.cpp

```diff
@@ -53,35 +53,23 @@ Shader::Shader(VkDevice device, const VkShaderStageFlagBits type, const std::str
     shader_module_ci.pCode = reinterpret_cast<const std::uint32_t *>(code.data());
 
     spdlog::debug("Creating shader module {}.", name);
-    if (vkCreateShaderModule(device, &shader_module_ci, nullptr, &m_shader_module) != VK_SUCCESS) {
+    if (vkCreateShaderModule(device.device(), &shader_module_ci, nullptr, &m_shader_module) != VK_SUCCESS) {
         throw std::runtime_error("Error: vkCreateShaderModule failed for shader " + name + "!");
     }
 
-    // Try to find the Vulkan debug marker function.
-    auto *vkDebugMarkerSetObjectNameEXT = reinterpret_cast<PFN_vkDebugMarkerSetObjectNameEXT>(
-        vkGetDeviceProcAddr(device, "vkDebugMarkerSetObjectNameEXT"));
-
-    if (vkDebugMarkerSetObjectNameEXT != nullptr) {
-        // Since the function vkDebugMarkerSetObjectNameEXT has been found, we can assign an internal name for
-        // debugging.
-        auto name_info = make_info<VkDebugMarkerObjectNameInfoEXT>();
-        name_info.objectType = VK_DEBUG_REPORT_OBJECT_TYPE_SHADER_MODULE_EXT;
-        name_info.object = reinterpret_cast<std::uint64_t>(m_shader_module);
-        name_info.pObjectName = name.c_str();
-
-        spdlog::debug("Assigning internal name {} to shader module.", name);
-        if (vkDebugMarkerSetObjectNameEXT(device, &name_info) != VK_SUCCESS) {
-            throw std::runtime_error("Error: vkDebugMarkerSetObjectNameEXT failed for shader " + name + "!");
-        }
-    }
+#ifndef NDEBUG
+    // Assign an internal name using Vulkan debug markers.
+    m_device.set_object_name((std::uint64_t)m_shader_module, VK_DEBUG_REPORT_OBJECT_TYPE_SHADER_MODULE_EXT, name);
+#endif
 }
 
-Shader::Shader(Shader &&shader) noexcept
-    : m_device(shader.m_device), m_type(shader.m_type), m_name(std::move(shader.m_name)),
-      m_entry_point(std::move(shader.m_entry_point)), m_shader_module(std::exchange(shader.m_shader_module, nullptr)) {}
+Shader::Shader(Shader &&other) noexcept
+    : m_device(other.m_device), m_type(other.m_type), m_name(std::move(other.m_name)),
+      m_entry_point(std::move(other.m_entry_point)), m_shader_module(std::exchange(other.m_shader_module, nullptr)) {}
 
 Shader::~Shader() {
-    vkDestroyShaderModule(m_device, m_shader_module, nullptr);
+    spdlog::trace("Destroying shader module {}.", m_name);
```

Comment:
We really need to start putting `if`s around these log messages (or just get rid of them completely).

Reply:
I will do this in another pull request: TODO https://github.com/inexorgame/vulkan-renderer/issues/225.